### ISSUE TITLE
Multiple Slow Queries Being Submitted By Metabase (Not Canceling Prior Ones Correctly)

### ISSUE THREAD

**Issue by orware** [metabase/metabase#5373]
### Bugs
- Your browser and the version: Latest Chrome
- Your operating system: Windows 7, Windows 10
- Your databases: Oracle
- Metabase version: 0.24.1 (however, I've just updated to 0.24.2 but we can't retest the issue exactly at this time)
- Metabase hosting environment: Windows Server 2016
- Metabase internal database: MariaDB 10.2.6

This is partly related to #5133 which I feel needs to still be reopened since it's a valid request, but this is an entirely different we came across yesterday.

It's related to #5133, because rather than the Researcher on our end just being able to run their query normally, I've asked them to wrap their query inside of the other query using `ROWNUM` shown below since that's the method of doing a simple LIMIT in Oracle:
```
select *
from  
(select * 
  from sv_student_term_data) 
where ROWNUM <= 5
```

However, this report is being made because the query above shouldn't have really caused an issue but it did because the `SV_STUDENT_TERM_DATA` view is pretty slow to begin with, so normally, it's a query you have to wait for a while on as it chugs through the data (in our primary reporting tool, evisions' Argos, a report that uses this view normally takes ~6 minutes to run).

Our researcher was submitting this query using Metabase though, and seeing that it wasn't loading anything (basically, it seemed to go back to the "If you give me some data I can show you something cool. Run a Query!" message after a few seconds) he was retrying things multiple times.

On the Oracle side though, it appeared that the slow query was getting submitted multiple times as well, and this was causing a really heavy load after some time had passed and quite a few of these queries had gotten accumulated.

So the bug it seems, is that for a slower query, Metabase may not be correctly canceling a query when it's no longer needed, and may instead be keeping queries alive that aren't even going to get used by anything in the web UI (e.g. I'm pretty sure the web UI would only be waiting for the latest request he had submitted, rather than any of the earlier ones).

To put this in perspective from the screenshot our DBA sent me of what he saw in Oracle, our normal usage would have been around 2 CPU cores worth of load, and from noon to 1 PM yesterday, it steadily climbed to about 20 cores and it had multiple processes using 5% of CPU each that all were coming from Metabase and the only query in the logs I could see multiple times was the single one I shared above (and this researcher is my only user testing Metabase out and using it currently).

I'm not sure how you guys could easily replicate this issue on your end, but it definitely caused us problems yesterday and so I hope there's a way for it to be fixed in Metabase because it was definitely triggered by how the software managed the queries it was starting up.

**Comment by camsaul** [comment]
Sounds like a duplicate of #2399

**Comment by orware** [comment]
@camsaul In some ways yes it's pretty much the same as that one...I think the main difference here is that the web UI wasn't even saying that it would take too long, it just seems to be making it seem like the query didn't execute properly or something. For example, if I try running the same query above in SQL Developer, it's taking a while and I'm still waiting for it to finish right now, but at least I have an indication that it's still running. In Metabase it's not doing that and goes back to the "If you give me some data I can show you something cool. Run a Query!" message, making it really easy for users to do the wrong thing and resubmit the query again (causing the server load build up we noticed, since Metabase was secretly keeping the past queries/connections opened it seems).

**Comment by dannyeuu** [comment]
@salsakran Any update here? how to prevent Jetty locks thread?

**Comment by camsaul** [comment]
Fixed by #11832. I implemented custom low-level JDBC code that avoids transactions altogether and is much more aggressive about cleaning up resources and canceling queries.

**Comment by orware** [comment]
Great news @camsaul! Thank you for the update and resolution for this particular issue :-)!
